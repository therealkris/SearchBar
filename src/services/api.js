// @flow

const RESULTS = ["Random", "Real", "Record", "Red", "Report"];

export const suggestTerms = (term: string) => {
  return RESULTS.filter(result => {
    return result.toLowerCase().includes(term.toLowerCase());
  });
};

export const search = (term: string) => {
  return {
    result: `You searched for ${term}`
  };
};
