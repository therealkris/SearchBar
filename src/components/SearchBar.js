// @flow
import * as React from "react";

import { suggestTerms, search } from "../services/api";
import Results from "./Results";
import styles from './styles.module.scss'

type SearchState = {
  results: Array<string>,
  term: string
};

export default class SearchBar extends React.Component<{}, SearchState> {
  state = {
    results: [],
    term: ""
  };

  handleChange = async e => {
    const value = e.target.value;
    if (value.length > 0) {
      const results = await suggestTerms(value);
      this.setState({ results, term: value });
    } else {
      this.setState({ results: [], term: value });
    }
  };

  handleClick = async (term: string) => {
    const results = await search(term);
    console.log(results);
  };

  render() {
    const { results, term } = this.state;
    return (
      <div className={styles.search}>
        <input
          type="search"
          name="search"
          onChange={this.handleChange}
          placeholder={"Start typing to search"}
        />
        {results.length > 0 && (
          <Results results={results} term={term} handleClick={this.handleClick} />
        )}
      </div>
    );
  }
}
