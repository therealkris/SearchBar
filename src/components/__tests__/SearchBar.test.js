import React from "react";
import { render, fireEvent } from "react-testing-library";
import "jest-dom/extend-expect";

import SearchBar from "../SearchBar";
import { suggestTerms } from "../../services/api";

jest.mock("../../services/api", () => {
  return { suggestTerms: jest.fn(() => Promise.resolve([])) };
});

test("renders correctly", () => {
  const { container } = render(<SearchBar />);
  expect(container).toMatchSnapshot();
});

test("handles change", () => {
  const { getByPlaceholderText } = render(<SearchBar />);
  const inputNode = getByPlaceholderText("Start typing to search");
  fireEvent.change(inputNode, { target: { value: "re" } });
  expect(suggestTerms).toHaveBeenCalledTimes(1);
});
