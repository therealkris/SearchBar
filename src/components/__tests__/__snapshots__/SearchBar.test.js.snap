// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <input
      name="search"
      placeholder="Start typing to search"
      type="search"
    />
  </div>
</div>
`;
